internal/scheduler/database/job_repository.go

```diff
@@ -73,6 +79,71 @@ func NewPostgresJobRepository(db *pgxpool.Pool, batchSize int32) *PostgresJobRep
 	}
 }
 
+func (r *PostgresJobRepository) FetchInitialJobs(ctx *armadacontext.Context) ([]Job, []Run, error) {
+	var updatedJobs []Job = nil
+	var initialRuns []Run = nil
+
+	start := time.Now()
+	defer func() {
+		ctx.Infof(
+			"received %d initial jobs and %d initial job runs from postgres in %s",
+			len(updatedJobs), len(initialRuns), time.Since(start),
+		)
+	}()
+
+	// Use a RepeatableRead transaction here so that we get consistency between jobs and dbRuns
+	err := pgx.BeginTxFunc(ctx, r.db, pgx.TxOptions{
+		IsoLevel:       pgx.RepeatableRead,
+		AccessMode:     pgx.ReadOnly,
+		DeferrableMode: pgx.Deferrable,
+	}, func(tx pgx.Tx) error {
+		var err error
+		queries := New(tx)
+
+		// Fetch jobs
+		initialJobRows, err := fetch(0, r.batchSize, func(from int64) ([]SelectInitialJobsRow, error) {
+			return queries.SelectInitialJobs(ctx, SelectInitialJobsParams{Serial: from, Limit: r.batchSize})
+		})
+		updatedJobs = make([]Job, len(initialJobRows))
+		updatedJobIds := make([]string, len(initialJobRows))
+		for i, row := range initialJobRows {
+			updatedJobIds[i] = row.JobID
+			updatedJobs[i] = Job{
+				JobID:                   row.JobID,
+				JobSet:                  row.JobSet,
+				Queue:                   row.Queue,
+				Priority:                row.Priority,
+				Submitted:               row.Submitted,
+				Validated:               row.Validated,
+				Queued:                  row.Queued,
+				QueuedVersion:           row.QueuedVersion,
+				CancelRequested:         row.CancelRequested,
+				Cancelled:               row.Cancelled,
+				CancelByJobsetRequested: row.CancelByJobsetRequested,
+				Succeeded:               row.Succeeded,
+				Failed:                  row.Failed,
+				SchedulingInfo:          row.SchedulingInfo,
+				SchedulingInfoVersion:   row.SchedulingInfoVersion,
+				Serial:                  row.Serial,
+				Pools:                   row.Pools,
+			}
+		}
+
+		if err != nil {
```

Comment:
After looking at the `fetch` function, if it returns an error, the returned jobs will be `nil`.
Can we move the error check right after the `fetch` call?

Reply:
Yeah that is fair, again just copy pasted behavior